### DISCUSSION TITLE
Twitter Webhook - Yes, you can add image cover on Twitter!

### DISCUSSION THREAD

**Discussion by Mozitux** [AzuraCast/AzuraCast#5656] [Show and tell]
Hi ;

I saw that it was not natively possible to automatically add music cover images to the Twitter webhook. This functionality has already been proposed, it seems to me, but not yet implemented. So in the meantime I've created a really basic script that fetches the cover images of the current title and can post them to Twitter in addition to the title, artist, and all the other info that's available in the API like this:

![twitter_webhook](https://user-images.githubusercontent.com/19860756/183297698-37ab1c76-707b-48b3-8357-85834567331e.PNG)

I'm not sure if it's really allowed by Twitter to post images every X minutes, but if you want, I can make you an installation guide with the script in question! Let me know if you're interested, and I will work on version 2. :)

**Comment by BusterNeece** [comment]
@Mozitux Would you mind sharing your script for the folks here? I haven't had a chance to look into the Twitter post API to see how to natively implement this, so it may provide useful information for us to implement it directly inside AzuraCast.

**Reply by Mozitux** [reply]
@SlvrEagle23 Sure!
However, I don't use the Twitter post API directly.
I'm using a .PHP file that will retrieve the image of the current title and add it to the Twitter META card image of the script.

In the Twitter Webhook, you must then add the link to the page, and add the request plus the tag "`?={{ now_playing.played_at}}`" at the end of the URL so that Twitter considers it as a new page on each new tweet and does not add the preview of the previous image that it has in cache.

When a listener clicks on the page link on the tweet, there is simply a redirection to the desired web radio page.
It's the easiest and fastest I've found so far, but I'll look into using the post API directly.

Here are the script (I can't put it directly in GitHub comment):
[twitter-webhook_image grabber.zip]
 
Of course, if you want a smaller image on Twitter, you can replace this on line 10:  
`<meta name="twitter:card" content="summary_large_image">`
by this: `<meta name="twitter:card" content="summary">`

The result: 
![twitter_webhook2](https://user-images.githubusercontent.com/19860756/183315076-5ec63fe7-1915-45a5-990d-01b1132670f9.PNG)

**Reply by BusterNeece** [reply]
@Mozitux Ohh, I see what you're doing...you're just replacing the stock image that would appear via the OG metadata on the embedded page with the current album art.

That's certainly something we could look into doing...basically if you set "set album art as image" on the Twitter hook, we'd change the URL to indicate to us to generate the proper metadata on our end for that specific playback link.

**Reply by Mozitux** [reply]
@SlvrEagle23 Exactly, I wanted to make it simpler and faster to build, but if it allows you to integrate it into Azuracast, that's great!
